gerrit-checklist.js

```diff
@@ -1,76 +1,160 @@
 // ==UserScript==
-// @name       Gerrit Review Checklist
-// @namespace  https://github.com/mziwisky/gerrit-checklist
-// @version    0.0.1
-// @description  Add a checklist to your Gerrit review panel, and record responses with your submitted comments.  Tested only with the "new UI" of Gerrit version 2.8.3.
-// @match      https://gerrit.instructure.com/*
-// @copyright  2014+, Michael Ziwisky
+// @name         Gerrit Review Checklist 2.0
+// @namespace    https://github.com/mziwisky/gerrit-checklist
+// @version      2.0.0
+// @description  Add a checklist to your Gerrit review panel, and record responses with your submitted comments.
+// @match        https://gerrit.instructure.com/*
+// @copyright    2014+ Michael Ziwisky, 2019+ Kyle Grinstead
 // ==/UserScript==
 
-function ReviewChecklistManager(optsList) {
+// example .gerrit-checklist.json file
+// this file should be in the root directory of the repo
+//
+// {
+//   "checklist": [
+//     "FE: works in a mobile browser",
+//     "BE: queries are optimized"
+//   ],
+//  "defaultStatus": false // a default status of false == Not Checked, true == Checked, null == N/A
+// }
+
+function ReviewChecklistManager(optsList, defaultStatus) {
   console.log('Gerrit review checklist added!');
-  var mgr = this;
+  const mgr = this;
   this.options = optsList.map(function(opt) {
-    var option = {
+    const option = {
       text: opt,
-      status: null,
+      status: defaultStatus,
       setStatus: function(status) {
         this.status = status;
         mgr.updateManagedTextArea();
-      }
+      },
     };
     return option;
   });
-};
+  this.isPolyGerrit = !!document.querySelector('gr-app');
+  this.textAreaDomSelector = this.isPolyGerrit
+    ? 'gr-reply-dialog gr-textarea textarea'
+    : '.popupContent .gwt-TextArea';
+}
+
+function buildElement(tag, options) {
+  const element = document.createElement(tag);
+  Object.keys(options || {}).forEach(function(optionKey) {
+    if (options[optionKey] !== null) {
+      element[optionKey] = options[optionKey];
+    }
+  });
+  return element;
+}
 
-ReviewChecklistManager.create = function(optsList) {
+ReviewChecklistManager.create = function(defaultOptsList) {
   if (ReviewChecklistManager.instanceCreated) {
     console.log('Only a single Gerrit checklist is allowed.');
     return;
   }
   ReviewChecklistManager.instanceCreated = true;
 
-  new ReviewChecklistManager(optsList).activate();
+  const repoName = document.location.toString().match(/\/c\/(.*)\/\+/)[1];
+  fetch(
+    `https://gerrit.instructure.com/plugins/gitiles/${repoName}/+show/master/.review-checklist.json?format=text`,
```

Comment:
can this just be a relative path, `/plugins/gitiles/...`?  i know there are already several assumptions being made, but it would makes it ever so slightly more portable